src/assignmentNotification.js

```diff
@@ -20,13 +20,9 @@ AssignmentNotification.prototype.send = function() {
 
   const firstPersist = this._persistAssignment();
 
-  const secondPersist = new Promise((resolve, reject) => {
-    this._visitor.analytics.trackAssignment(this._visitor.getId(), this._assignment, success =>
-      this._persistAssignment(success ? 'success' : 'failure')
-        .then(resolve)
-        .catch(reject)
-    );
-  });
+  const secondPersist = this._visitor.analytics
+    .trackAssignment(this._visitor.getId(), this._assignment)
+    .then(success => this._persistAssignment(success ? 'success' : 'failure'));
```

Comment:
since this contract is changing, we'll need to update any apps that provide their own analytics

Reply:
we should update the docs too: https://github.com/Betterment/test_track_js_client#advanced-configuration

Reply:
Ah good catch. We _could_ do that since this is part of a major version bump as a breaking change. I'm also cool with keeping the callback API to make it easier to upgrade.

I can't speak to any advantage that `trackAssignment` using Promises would provide us over callbacks except for maybe standardizing on Promises internally.

Reply:
i did a little research: mixpanel, segment and kiss metrics all take a callback. i suspect that's because mixpanel takes a callback and everyone else is just following along.

i think for ease of integration with other analytics vendors, we should keep the callback.